Assets/Mirror/Examples/AdditiveScenes/Scripts/PlayerController.cs

```diff
@@ -5,28 +5,26 @@ namespace Mirror.Examples.Additive
     [RequireComponent(typeof(CharacterController))]
     public class PlayerController : NetworkBehaviour
     {
-        public override void OnStartServer()
-        {
-            base.OnStartServer();
-            playerColor = Random.ColorHSV(0f, 1f, 1f, 1f, 0.5f, 1f);
-        }
-
-        [SyncVar(hook = nameof(SetColor))]
-        public Color32 playerColor = Color.black;
+        public CharacterController characterController;
 
-        // Unity clones the material when GetComponent<Renderer>().material is called
-        // Cache it here and destroy it in OnDestroy to prevent a memory leak
-        Material cachedMaterial;
-
-        void SetColor(Color32 color)
+        void OnValidate()
         {
-            if (cachedMaterial == null) cachedMaterial = GetComponent<Renderer>().material;
-            cachedMaterial.color = color;
+            if (characterController == null)
+            {
+                // Auto-assign CharacterController.
+                characterController = GetComponent<CharacterController>();
+                if (characterController == null)
+                    characterController = gameObject.AddComponent<CharacterController>();
+            }
         }
 
-        void OnDestroy()
+        public override void OnStartLocalPlayer()
         {
-            Destroy(cachedMaterial);
+            base.OnStartLocalPlayer();
+
+            Camera.main.transform.SetParent(transform);
```

Comment:
Are you sure OnDisable is called when the player is destroyed?   because if not,  you will end up destroying the camera when the client disconnects.

And why is this code here,  shouldn't it be in a separate PR?  this seems to have nothing to do with refactoring the Color

Reply:
This code was always here, just the methods were moved around and GitHub doesn't show that well.